x/swaprouter/router_test.go

```diff
@@ -161,30 +539,79 @@ func (suite *KeeperTestSuite) TestMultihopSwapExactAmountOut() {
 	for _, tc := range tests {
 		suite.Run(tc.name, func() {
 			suite.SetupTest()
-
+			gammKeeper := suite.App.GAMMKeeper
 			swaprouterKeeper := suite.App.SwapRouterKeeper
 
-			suite.createBalancerPoolsFromCoins(tc.poolCoins)
+			suite.createBalancerPoolsFromCoinsWithSwapFee(tc.poolCoins, tc.poolFee)
 
-			// if we expect a reduced fee to apply, we set both pools in DistrInfo to replicate it being an incentivized pool
-			// each pool has three gauges, hence 6 gauges for 2 pools
-			if tc.reducedFeeApplied {
-				test := poolincentivestypes.DistrInfo{
-					TotalWeight: sdk.NewInt(2),
-					Records: []poolincentivestypes.DistrRecord{
-						{GaugeId: 1, Weight: sdk.OneInt()}, {GaugeId: 2, Weight: sdk.OneInt()}, {GaugeId: 3, Weight: sdk.OneInt()},
-						{GaugeId: 4, Weight: sdk.OneInt()}, {GaugeId: 5, Weight: sdk.OneInt()}, {GaugeId: 6, Weight: sdk.OneInt()}},
+			// if test specifies incentivized gauges, set them here
+			if len(tc.incentivizedGauges) > 0 {
+				var records []poolincentivestypes.DistrRecord
+				totalWeight := sdk.NewInt(int64(len(tc.incentivizedGauges)))
+				for _, gauge := range tc.incentivizedGauges {
+					records = append(records, poolincentivestypes.DistrRecord{GaugeId: gauge, Weight: sdk.OneInt()})
 				}
-				suite.App.PoolIncentivesKeeper.SetDistrInfo(suite.Ctx, test)
+				distInfo := poolincentivestypes.DistrInfo{
+					TotalWeight: totalWeight,
+					Records:     records,
+				}
+				suite.App.PoolIncentivesKeeper.SetDistrInfo(suite.Ctx, distInfo)
 			}
 
-			tokenIn, err := swaprouterKeeper.RouteExactAmountOut(suite.Ctx, suite.TestAccs[0], tc.routes, tc.tokenInMaxAmount, tc.tokenOut)
+			// calcOutAmountAsSeparateSwaps calculates the multi-hop swap as separate swaps, while also
+			// utilizing a cacheContext so the state does not change
+			calcOutAmountAsSeparateSwaps := func(osmoFeeReduced bool) sdk.Coin {
```

Comment:
Consider moving this function definition either to be defined at the top of the test fixture or as a separate helper function in the test-global scope.

Similar suggestion for the other test.

The reason is to keep the actual test simple and easy to follow

Reply:
Good call 👍